### PR TITLE
Document ShaderInclude

### PR DESCRIPTION
**Pull request by MewPurPur**
A comment

### CODE REVIEW

**Review by QbieShay — APPROVED** [review]
Content is good, not sure about form.

File: doc/classes/ShaderInclude.xml [hunk 1]
```diff
@@ -1,13 +1,17 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="ShaderInclude" inherits="Resource" version="4.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
+		A snippet of shader code to be included in a [Shader] with [code]#include[/code].
 	</brief_description>
 	<description>
+		A shader include file. Allows you to define a custom shader snippet that can be included in a [Shader] by using the preprocessor directive [code]#include[/code]. The snippet doesn't have to be a valid shader on its own.
```
Comment (akien-mga) [inline doc/classes/ShaderInclude.xml:7]:
```suggestion
		A shader include file, saved with the [code].gdshaderinc[/code] extension. Allows you to define a custom shader snippet that can be included in a [Shader] by using the preprocessor directive [code]#include[/code], followed by the file path (e.g. [code]#include "res://shader_lib.gdshaderinc"[/code]). The snippet doesn't have to be a valid shader on its own.
```

**Review by akien-mga — APPROVED** [review]

File: doc/classes/ShaderInclude.xml (hunk 1 above)
Comment (MewPurPur) [reply doc/classes/ShaderInclude.xml:7]: Hmm, okay. We don't mention what the extension is in any other similar classes; I'll do this here and change the rest in a follow-up.

Comment (akien-mga) [reply doc/classes/ShaderInclude.xml]: I'm not necessarily suggesting such a change for all Resources. This one is special in that it's a resource matching a code file that needs to be saved with this extension and used like this. Other resources just get `load`ed in scripts.

Comment (MewPurPur) [reply doc/classes/ShaderInclude.xml]: I meant GDScript, CSharpScript (maybe?), Shader.

Comment (akien-mga) [reply doc/classes/ShaderInclude.xml]: Yeah it would make sense for those 👍

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.4.

**Pull request merged**